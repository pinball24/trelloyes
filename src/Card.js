import React from 'react';
import './Card.css';


class Card extends React.Component {
    deleteCard(event) {
        event.preventDefault();
        this.props.onDeleteItem(this.props.itemId);
    }
    render() {
        return (
            <div className="Card">
                <button 
                    onClick={e => this.deleteCard(e)}
                    type="button">
                    delete
                </button>
                <h3>{this.props.title}</h3>
                <p>{this.props.content}</p>
            </div>
        )
    }
}

export default Card;
