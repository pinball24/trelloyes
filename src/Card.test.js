import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<Card title="Hello" content="foo" />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('renders UI as expected', () => {
    const tree = renderer
        .create(<Card title="Hello" content="foo" />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

