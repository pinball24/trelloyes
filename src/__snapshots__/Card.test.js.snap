// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    Hello
  </h3>
  <p>
    foo
  </p>
</div>
`;
